DashAI/back/plugins/utils.py

```diff
@@ -68,3 +76,79 @@ def get_plugins_from_pypi() -> List[dict]:
         if plugin_name.lower().startswith("dashai") and plugin_name.lower() != "dashai"
     ]
     return [_get_plugin_by_name_from_pypi(plugin_name) for plugin_name in plugins_names]
+
+
+def available_plugins() -> List[type]:
+    """
+    Get available DashAI plugins entrypoints
+
+    Returns
+    ----------
+    List[type]
+        A list of plugins' classes
+    """
+    # Retrieve plugins groups (DashAI components)
+    plugins = entry_points(group="dashai.plugins")
+
+    # Look for installed plugins
+    plugins_list = []
+    for plugin in plugins:
+        # Retrieve plugin class
+        plugin_class = plugin.load()
+        plugins_list.append(plugin_class)
+
+    return plugins_list
+
+
+def register_new_plugins(component_registry) -> List[type]:
```

Comment:
I suggest to take the avialable plugins as parameter and not to execute the function `avialable_plugins` inside this one.
(check single responsability principle + inversion of control in software).

This implies that the function or method that executes this action should call get_available_plugins before this function.

```suggestion
def register_new_plugins(component_registry: ComponentRegistry, available_plugins: List[type]) -> List[type]:
```

Reply:
Should we do the same with the registered component classes? Something like

```python
def get_registered_component_classes(component_registry: ComponentRegistry) -> List[type]:
    registered_components = []
        for component in component_registry.get_components_by_types():
            registered_components.append(component["class"])
        return registered_components
```

Reply:
This change: register_new_plugins(component_registry: ComponentRegistry, available_plugins: List[type]) is ready

Reply:
> Should we do the same with the registered component classes? Something like

I think it is not necessary because what you are doing is according to the responsibility of the function, (get the classes from the component registry, not operate on them).

Also, you could abbreviate this function by using:

```python
def get_registered_component_classes(component_registry: ComponentRegistry) -> List[type]:
    return [component["class"] for component in component_registry.get_components_by_types()]
```

Reply:
Ready this changes!